test/e2e_env/universal/auth/dp_auth.go

```diff
@@ -51,18 +51,19 @@ func DpAuth() {
 		dp := builders.Dataplane().
 			WithName("dp-01").
 			WithMesh(meshName).
+			WithLabels(map[string]string{"app": "not-test-server"}).
 			WithAddress("192.168.0.2").
 			WithServices("test-server").
 			Build()
 		Expect(universal.Cluster.Install(ResourceUniversal(dp))).To(Succeed())
 
 		// when
-		err := TestServerUniversal("dp-02", meshName, WithServiceName("test-server"))(universal.Cluster)
+		err := TestServerUniversal("dp-02", meshName, WithServiceName("test-server"), WithAppLabel("test-server"))(universal.Cluster)
 		Expect(err).ToNot(HaveOccurred())
 
 		// then
 		Eventually(func() (string, error) {
 			return universal.Cluster.GetKumactlOptions().RunKumactlAndGetOutput("get", "dataplanes", "-oyaml")
-		}, "30s", "1s").ShouldNot(ContainSubstring("192.168.0.2"))
+		}, "30s", "1s").ShouldNot(And(ContainSubstring("192.168.0.2"), ContainSubstring("not-test-server")))
```

Comment:
Please check if it actually contains `app: test-server`, otherwise this test would'be passed before your fix

Reply:
updated asssertion